Serenity-Website/src/Journal/JournalPage.jsx

```diff
@@ -0,0 +1,336 @@
+import "./JournalPage.css";
+
+import react, { useState, useEffect } from "react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import {
+  faStar,
+  faThumbsUp,
+  faThumbsDown,
+} from "@fortawesome/free-regular-svg-icons";
+import ToolTip from "../ToolTip/ToolTip";
+import { useSidebar } from "../sidebarcontext/SidebarContext";
+import Header from "../Header/Header";
+import LoadingScreen from "../Loading-Spinner/LoadingSpinner";
+
+const JournalEntry = () => {
+  const [prompt, setPrompt] = useState("");
+  const [content, setContent] = useState("");
+
+  // stores the journal ID from the request body
+  const [journalId, setJournalId] = useState(null);
+
+  // tracks if user marks journal as favorite
+  const [isFavorite, setIsFavorite] = useState(false);
+
+  // Tracks whether the user has submitted the journal entry (true) or not (false).
+  const [isSubmitted, setIsSubmitted] = useState(false);
+
+  // Controls the edit mode state for the journal entry, toggling between read-only and editable states.
+  //When set to true, enables typing in the text area for editing journal content.
+  const [isEditing, setIsEditing] = useState(false);
+
+  // state hooks for upvote/downvote status
+  const [upvotes, setUpVotes] = useState(0);
+  const [downvotes, setDownVotes] = useState(0);
+  const [userUpVoted, setUserUpVoted] = useState(false);
+  const [userDownVoted, setUserDownVoted] = useState(false);
+
+  const { isSidebarOpen, toggleSidebar } = useSidebar();
+
+  // state hook for loading ui
+  const [loading, setLoading] = useState(false);
+
+  const FavoriteIcon = <FontAwesomeIcon icon={faStar} />;
+
+  const backendUrl = import.meta.env.VITE_BACKEND_ADDRESS;
+  const promptUrl = `${backendUrl}/generateJournalPrompt`;
+  const createJournalUrl = `${backendUrl}/create-journal-entry`;
+  const updateJournalUrl = `${backendUrl}/update-journal-entry`;
+  const favoriteJournalUrl = `${backendUrl}/journal/favorite`;
+  const votesUrl = `${backendUrl}/journal/vote`;
+
+  useEffect(() => {
```

Comment:
Discussed offline, but just adding the note here for others:
- This hook will result in the Post request running everytime

Sounds like this fix will be done afterwards

Reply:
Yes, I will implement it in my next PR.